sgl-kernel/csrc/moe/moe_reorder_kernel.cu

```diff
@@ -0,0 +1,68 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+
+#include <THC/THCAtomics.cuh>
+
+#include "utils.h"
+
+__global__ void pre_reorder_cuda_kernel(
+    const float* __restrict__ input_ptr,
+    float* __restrict__ gateup_input_ptr,
+    const int* __restrict__ src2dst_ptr,
+    const int* __restrict__ topk_ids_ptr,
+    const float* __restrict__ a1_scales_ptr,
+    int start_expert_id,
+    int end_expert_id,
+    int topk,
+    int hidden_size) {
+  int token_idx = blockIdx.x;
+  int tid = threadIdx.x;
+
+  const float* src_ptr = input_ptr + token_idx * hidden_size;
+  const int* token_src2dst = src2dst_ptr + token_idx * topk;
+  const int* token_topk_ids = topk_ids_ptr + token_idx * topk;
+
+  for (int k = 0; k < topk; ++k) {
+    int expert_id = token_topk_ids[k];
+    if (expert_id < start_expert_id || expert_id > end_expert_id) continue;
+
+    float scale = 1.0f;
+    if (a1_scales_ptr != nullptr) {
+      scale = 1.0f / a1_scales_ptr[expert_id - start_expert_id];
+    }
+
+    int dst_idx = token_src2dst[k];
+    float* dst_ptr = gateup_input_ptr + dst_idx * hidden_size;
+
+    for (int i = tid; i < hidden_size; i += blockDim.x) {
+      dst_ptr[i] = src_ptr[i] * scale;
+    }
+  }
+}
+
+void moe_pre_reorder(
```

Comment:
For the input Tensor, the function signature does not need to add the suffix '_ptr'. This is also consistent with the code in sgl-kernel. Can we make this change?

Reply:
Sure. Changed.